esx_service/utils/auth_api.py

```diff
@@ -973,24 +1021,28 @@ def _tenant_access_rm(name, datastore):
     if error_info:
         return error_info
 
-    logging.debug("_tenant_access_rm: datastore_url=%s", datastore_url)
-    error_msg = tenant.remove_datastore_access_privileges(auth_mgr.conn, datastore_url)
-    if error_msg:
-        error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
-        return error_info
-
-    # get default_datastore, if default_datastore is the same as param "datastore"
-    # need to set default_datastore_url to "" in tenants table
+    # get dafault_datastore for this tenant
+    # if the default_datastore is equal to param "datastore", which means
+    # we are trying to remove a row in "privilege" table with datastore which is
+    # marked as default_datastore of this tenant, should return with error
     error_info, default_datastore_url = get_default_datastore_url(name)
     if error_info:
         return error_info
 
     if default_datastore_url == datastore_url:
-        error_msg = tenant.set_default_datastore(auth_mgr.conn, "")
-        if error_msg:
-            error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
+        msg = "{} is set as default datastore".format(datastore)
```

Comment:
Why the "create" should be "sticky" on default_ds? Any use cases here? The check in "_tenant_access_rm" just want to make sure it must have privilege to "default_datastore" for a tenant. Why should we enforce that it must allow "create" here?